internal/remote/sources.go

```diff
@@ -0,0 +1,133 @@
+package remote
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	core "github.com/speakeasy-api/speakeasy-core/auth"
+	"github.com/speakeasy-api/speakeasy/internal/sdk"
+
+	"github.com/speakeasy-api/speakeasy-client-sdk-go/v3/pkg/models/operations"
+	"github.com/speakeasy-api/speakeasy-client-sdk-go/v3/pkg/models/shared"
+)
+
+// RecentGeneration represents a recent generation of a target in the workspace.
+// The source of this data is our CLI event stream, which is updated every time
+// a target is generated.
+type RecentGeneration struct {
+	CreatedAt            time.Time
+	ID                   string
+	TargetName           string
+	Target               string
+	SourceNamespace      string
+	SourceRevisionDigest string
+	Success              bool
+	Published            bool
+
+	// May not be set
+	GitRepoOrg *string
+	GitRepo    *string
+
+	// gen.yaml
+	GenerateConfig *string
+}
+
+const (
+	// The event stream contains multiple events for the same namespace, so we want to
+	// break execution once we've seen a minimum, arbitrary number of unique namespaces.
+	recentGenerationsToShow int = 5
+)
+
+// GetRecentWorkspaceGenerations returns the most recent generations of targets in a workspace
+// This is based on the CLi event stream, which is updated on every CLI interaction.
+func GetRecentWorkspaceGenerations(ctx context.Context) ([]RecentGeneration, error) {
```

Comment:
I think we should look to make this smart. We should prioritize targets that are actually connected to Github or are connected to publishing. We may want to consider only including these targets

Once someone is already onboarded we probably don't want people basing new SDKs off of random local generations on someone's machine, I think that will lead to problems because these specs may never be updated again.